relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC

Steps to reproduce: clone and `make`.

```
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/html/html.o src/html/html.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/html/houdini_href_e.o src/html/houdini_href_e.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/html/houdini_html_e.o src/html/houdini_html_e.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/html/houdini_html_u.o src/html/houdini_html_u.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/inlines.o src/inlines.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/buffer.o src/buffer.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/blocks.o src/blocks.c
re2c --case-insensitive -bis src/scanners.re > src/scanners.c || (rm src/scanners.c && false)
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/print.o src/print.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/utf8.o src/utf8.c
cc -g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC    -c -o src/references.o src/references.c
cc -g -O3 -Wall -Werror  -o cmark src/html/html.o src/html/houdini_href_e.o src/html/houdini_html_e.o src/html/houdini_html_u.o src/inlines.o src/buffer.o src/blocks.o src/scanners.c src/print.o src/utf8.o src/references.o src/main.c
cc -g -O3 -Wall -Werror  -shared -o libcmark.so src/html/html.o src/html/houdini_href_e.o src/html/houdini_html_e.o src/html/houdini_html_u.o src/inlines.o src/buffer.o src/blocks.o src/scanners.c src/print.o src/utf8.o src/references.o
/usr/lib/gcc/x86_64-pc-linux-gnu/4.7.3/../../../../x86_64-pc-linux-gnu/bin/ld: /tmp/cc9k94ZI.o: relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
/tmp/cc9k94ZI.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:72: recipe for target 'libcmark.so' failed
make: *** [libcmark.so] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 CFLAGS?=-g -O3 -Wall -Wextra -std=c99 -Isrc -Wno-missing-field-initializers -fPIC $(OPTCFLAGS)
-LDFLAGS?=-g -O3 -Wall -Werror $(OPTLDFLAGS)
+LDFLAGS?=-g -O3 -Wall -Werror -fPIC $(OPTLDFLAGS)
 SRCDIR?=src
 DATADIR?=data
 BENCHINP?=README.md

